tests/hardware_raspyrfm.c

```diff
@@ -0,0 +1,321 @@
+	data->origin = 1;
+	data->rawlen = ARRAY_LEN(txtestpulses);
+	memcpy(data->pulses, txtestpulses, data->rawlen*sizeof(int));
+	data->txrpt = 0;
+	strncpy(data->protocol, "dummy", 255);
+	data->hwtype = 1;
+	memset(data->uuid, 0, UUID_LENGTH+1);
+	eventpool_trigger(REASON_SEND_CODE, reason_send_code_free, data);
+}
+
+static int start(const char *func, uv_connection_cb cb) {
+	if(wiringXSetup("test", foo) != -999) {
+		printf("[ %-31.31s (preload libgpio)]\n", __FUNCTION__);
+		fflush(stdout);
+		wiringXGC();
+		return -1;
+	}
+	printf("[ %-48s ]\n", func);
+	fflush(stdout);
+
+	memtrack();
+	
+	//create SPI dummy
+	uv_fs_t *req = malloc(sizeof(uv_fs_t));
+	uv_fs_open(uv_default_loop(), req, "/dev/spidev0.0", O_CREAT, O_RDWR, open_cb);
+
+	//create dummy config
+	FILE *f = fopen("hardware_raspyrfm.json", "w");
+	fprintf(f,
+		"{\"devices\":{},\"gui\":{},\"rules\":{},"\
+		"\"settings\":{\"gpio-platform\":\"gpio-stub\"},"\
+		"\"hardware\":{},\"registry\":{}}"
+	);
+	fclose(f);
+
+	//create GPIO dummy
+	int r = 0;
+
+	pipe_req = MALLOC(sizeof(uv_pipe_t));
+	CuAssertPtrNotNull(gtc, pipe_req);
+
+	r = uv_pipe_init(uv_default_loop(), pipe_req, 1);
+	CuAssertIntEquals(gtc, 0, r);
+
+	uv_fs_t file_req;
+	uv_fs_unlink(uv_default_loop(), &file_req, "/dev/gpio1", NULL);
+
+	r = uv_pipe_bind(pipe_req, "/dev/gpio1");
+	CuAssertIntEquals(gtc, 0, r);
+
+	r = uv_listen((uv_stream_t *)pipe_req, 128, cb);
+	CuAssertIntEquals(gtc, 0, r);
+
+	plua_init();
+	test_set_plua_path(gtc, __FILE__, "hardware_raspyrfm.c");
+
+	storage_init();
+	CuAssertIntEquals(gtc, 0, storage_read("hardware_raspyrfm.json", CONFIG_SETTINGS));
+
+	hardware_init();
+	raspyrfmInit();
+
+	raspyrfm->minrawlen = ARRAY_LEN(rxtestpulses);
+	raspyrfm->maxrawlen = ARRAY_LEN(rxtestpulses);
+
+	timer_req = MALLOC(sizeof(uv_timer_t));
+	CuAssertPtrNotNull(gtc, timer_req);
+	uv_timer_init(uv_default_loop(), timer_req);
+	uv_timer_start(timer_req, (void (*)(uv_timer_t *))stop, 1500, 0);
+
+	itestraw = 0;
+
+	uv_thread_create(&pth, initthread, NULL);
+
+	uv_run(uv_default_loop(), UV_RUN_DEFAULT);
+	uv_walk(uv_default_loop(), walk_cb, NULL);
+
+	while(uv_loop_close(uv_default_loop()) == UV_EBUSY) {
+		uv_run(uv_default_loop(), UV_RUN_ONCE);
+	}
+
+	uv_thread_join(&pth);
+
+	storage_gc();
+	plua_gc();
+	hardware_gc();
+	eventpool_gc();
+	wiringXGC();
+
+	CuAssertTrue(gtc, check == 1);
+	CuAssertIntEquals(gtc, 0, xfree());
+	
+	return 0;
+}
+
+static void test_hardware_raspyrfm_receive(CuTest *tc) {
+	gtc = tc;
+
+	eventpool_init(EVENTPOOL_NO_THREADS);
+	eventpool_callback(REASON_RECEIVED_PULSETRAIN, receivePulseTrain, 0);
+
+	if (start(__FUNCTION__, connect_receive_cb) == -1)
+		return;
+}
+
+static void test_hardware_raspyrfm_send(CuTest *tc) {
+	gtc = tc;
+
+	eventpool_init(EVENTPOOL_THREADED);
+
+	if (start(__FUNCTION__, connect_send_cb) == -1)
+		return;
+}
+
+CuSuite *suite_hardware_raspyrfm(void) {
+	CuSuite *suite = CuSuiteNew();
+
+	SUITE_ADD_TEST(suite, test_hardware_raspyrfm_send);
```

Comment:
Can you also add the tests that check the config parameters. As is check for valid and invalid parameters.